README.md

```diff
@@ -1,153 +1,164 @@
 
+* **openapi**: the OpenAPI 3 specification of the APIs exposed
 
-## API
+## GUI/dashboard
 
-Il server eroga delle API sia pubbliche che private.
-Quelle pubbliche sono descritte nelle [specifiche OpenAPI 3](openapi/anpr-dashboard.yaml).
+The application also exposes a web-based GUI that can be accessed from website root ('/') once the server is running.
 
-E' possibile recuperare lo stato dei comuni sia puntualmente
-che in maniera massiva.
+Through the GUI it's possible to view the main statistical data, download a csv file with the latest datasets and to upload a csv with new data.
 
-Un esempio di processamento dei dati è disponibile nel
-[file di esempio](openapi/migrazione-anpr.py)
+## APIs
 
-## Alcune query da eseguire sul server
+The service exposes both some public, as well some private APIs. The public ones are described in the [OpenAPI 3 specification](openapi/anpr-dashboard.yaml).
 
-Cerca fornitore per un COMUNE
+Through the APIs, it's -for example- possible to retrieve the state of the migration to ANPR for single a municipality or for all of them together.
 
-```
-SELECT C.NAME,F.NAME as NOME_FORNITORE FROM COMUNE C
-INNER JOIN FORNITORE F ON F.ID = C.ID_FORNITORE
-WHERE UPPER(C.NAME)  LIKE  "%CARATE BRIANZA%"```
+## Sandbox environments
+
+For development purposes the project can also be run locally, directly on the developer machine, or in form of a Docker container. Following, both procedures are explained.
+
+### Run the project directly on the local machine
+
+```shell
+# Configuration
+cd anpr-dashboard-server/server
+mkdir creds
+cp {THREE-CONFIG-FROM-ADMINS} creds
+mkdir dashboard-subentro
+cp {ANOTHER-CONFIG-FILE-FROM-ADMINS} subentroconfig.yaml dashboard-subentro/
+
+# Build the project
+make build
 
-SELECT ID,NAME FROM FORNITORE WHERE  UPPER(FORNITORE.NAME) LIKE "%SYSTEM%"
+# Run the service
+./server --mode=debug --config-file=/<PATH-TO-YOUR-WORKSPACE>/anpr-dashboard-server/dashboard-subentro/subentroconfig.yaml
 ```
 
-Cambia fornitore per un comune
+### Run the project as a Docker container
 
+A `Dockerfile` and a `docker-compose.yaml` files are in the root of this repository.
+
+Create a new *vault* directory in the root of this repository. Put your development configurations and credentials in it. Make sure they reference */srv/db/sqlite.db*. Put the sqlite database in *vault/db* and name it *sqlite.db*
+
+To build the local test environment run:
+
+```shell
+UID=$UID docker-compose up -d
 ```
-UPDATE COMUNE SET ID_FORNITORE = (SELECT ID FROM FORNITORE WHERE  UPPER(FORNITORE.NAME)="AP SYSTEMS") WHERE COMUNE.NAME= "CARATE BRIANZA"
+
+The website and the APIs should now be accessible on port *8080*. While the website is accessible at the root, the APIs can be accessed under the path */api*
+
+To bring down the test environment and remove the containers use
+
+```shell
+docker-compose down
 ```
 
+## Query examples
```

Comment:
@mccalv le query possiamo metterle in un altro file? Ha senso?

Reply:
@mccalv @ioggstream ok, le metto in un file a parte e faccio un link nel readme principale